### PR TITLE
Zwane/android load

### PR DESCRIPTION
**Pull request by zwanesl**
This change introduces running of an android version of 'stressapptest' which can be used for load & stress  (cpu/memory) testing of android hypervisors

### CODE REVIEW

File: lib/tests.js [hunk 1]
```diff
@@ -228,6 +228,25 @@ tests.androidLongTest = async function (driver, caps) {
...
 
+tests.androidLoadTest = async function (driver, caps) {
+  let testSeconds = 600;
+  let args = await driver.elementById('textOptions');
+  let goBtn = await driver.elementById('go');
+
+  await args.clear();
+  await args.sendKeys("-M 500 -v 20" + " -s " + testSeconds);
+
+  for (let test_minutes = testSeconds / 60; test_minutes > 0; test_minutes--) {
```
Comment (jlipps) [inline lib/tests.js]: `testMinutes` (camelCase)

File: lib/tests.js [hunk 2]
```diff
@@ -228,6 +228,25 @@ tests.androidLongTest = async function (driver, caps) {
...
+    await driver.sleep(30000);
```
Comment (jlipps) [inline lib/tests.js]: might be nice to have a comment to explain the sleeps, like `// here the app is doing its stressy thing so we wait`.

also, why is the sleep _before_ you click the button? doesn't that just add 30s of wait time up front?

File: lib/tests.js [hunk 3]
```diff
@@ -228,6 +228,25 @@ tests.androidLongTest = async function (driver, caps) {
...
+    await goBtn.click();
+
+    await driver.sleep(30000);
+    await goBtn.click();
+  }
+
+  await driver.closeApp();
```
Comment (jlipps) [inline lib/tests.js]: no need to call `closeApp`, since runsauce always calls `driver.quit` after each test anyway.

File: lib/caps.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ const APPS = {
   'iOSHybridWK': 'http://appium.s3.amazonaws.com/FLWebView.app.zip',
   'Android': 'http://appium.s3.amazonaws.com/ContactManager.apk',
   'AndroidHybrid': 'http://appium.s3.amazonaws.com/ApiDemos-debug-2015-03-19.apk',
+  'AndroidLoad': 'http://www.saucelabs.com/example_files/stressapptest.apk',
```
Comment (jlipps) [inline lib/caps.js]: can we put this up on s3 somewhere? probably more reliable than the sauce website?

File: lib/tests.js (hunk 1 above)
Comment (zwanesl) [reply lib/tests.js]: Of course!

File: lib/tests.js (hunk 3 above)
Comment (zwanesl) [reply lib/tests.js]: Thanks, I'll remove that

File: lib/tests.js (hunk 2 above)
Comment (zwanesl) [reply lib/tests.js]: Good point, I've cleaned that up. I've also added a comment

File: lib/caps.js (hunk 4 above)
Comment (zwanesl) [reply lib/caps.js]: I'll get that moved and updated.

**Comment by zwanesl** [comment]
The move to S3 is still pending...

File: lib/tests.js [hunk 5]
```diff
@@ -228,6 +228,25 @@ tests.androidLongTest = async function (driver, caps) {
...
+tests.androidLoadTest = async function (driver, caps) {
+  let testSeconds = 600;
+  let args = await driver.elementById('textOptions');
+  let goBtn = await driver.elementById('go');
+
+  await args.clear();
+  await args.sendKeys("-M 500 -v 20" + " -s " + testSeconds);
+
+  for (let testMinutes = testSeconds / 60; testMinutes > 0; testMinutes--) {
+    // Only the first click triggers the test, the rest of the clicks are ignored
```
Comment (jlipps) [inline lib/tests.js]: i see what you're trying to do here now; i think we can simplify this a bit (and don't forget template strings):

```js
const iterations = 20; // run for 10 minutes
const intervalMs = 30000;
...
await args.sendKeys(`-M 500 -v 20 -s ${Math.floor(iterations * intervalMs / 1000)}`);

for (let i = 0; i < iterations; i++) {
  // comment
  await goBtn.click();
  await driver.sleep(intervalMs);
}
```

Comment (zwane) [reply lib/tests.js]: Thanks that looks cleaner

**Comment by jlipps** [comment]
thanks @zwanesl, published as version 2.1.0

**Pull request merged**